elasticdl/python/master/master.py

```diff
@@ -103,18 +103,14 @@ def run(self):
         """
         try:
             while True:
-                if self.task_manager and self.task_manager.finished():
-                    if self.pod_manager:
-                        self.pod_manager.update_status(
-                            PodManagerStatus.FINISHED
-                        )
-                    break
                 if self.pod_manager and self.pod_manager.all_workers_exited:
-                    if self.task_manager:
+                    if self.task_manager and not self.task_manager.finished():
```

Comment:
If workers early stop, there will be tasks left. Thus, warning only.

Reply:
Done